dev/src/write-batch.ts

```diff
@@ -900,16 +903,8 @@ function validateUpdateMap(
   if (!isPlainObject(obj)) {
     throw new Error(customObjectMessage(arg, obj));
   }
-
-  let isEmpty = true;
-  if (obj) {
-    for (const prop of Object.keys(obj)) {
-      isEmpty = false;
-      validateFieldValue(arg, obj[prop], allowUndefined, new FieldPath(prop));
-    }
-  }
-
-  if (isEmpty) {
+  if (Object.keys(obj).length === 0) {
     throw new Error('At least one field must be updated.');
   }
+  validateFieldValue(arg, obj, allowUndefined);
```

Comment:
Can you explain why we no longer validate each field on the object anymore?

Reply:
We do. We loop through the object here - https://github.com/googleapis/nodejs-firestore/blob/56355f1c83b055fdbd3ed47e069df813696a1fc0/dev/src/serializer.ts#L334